tutorials/text/text_props.py

```diff
@@ -204,14 +212,27 @@
 # Korean, or Japanese.
 #
 # To set the default font to be one that supports the code points you
-# need, prepend the font name to ``'font.family'`` or the desired alias
-# lists ::
+# need, prepend the font name to ``'font.family'`` (recommended), or to the
+# desired alias lists (not recommended). ::
+#
+#    # recommended
+#    matplotlib.rcParams['font.family'] = ['Source Han Sans TW', 'sans-serif']
+#
+#    # not recommended
```

Comment:
This is a bit confusing, because this pattern is exactly what our default values are!

Reply:
And those default values are exactly what we don't want users to touch, if they want custom font families, go ahead and modify `font.family`, what we shouldn't recommend is to modify `font.serif`, `font.fantasy`, etc.

Reply:
I guess there is a hypothetical workflow where people are setting `matplotlib.rcParams['font.family'] = ['serif']` or `matplotlib.rcParams['font.family'] = ['cursive']`.  People could do

```
matplotlib.rcParams['font.sans-serif'] = ['Source Han Sans TW', ...]
matplotlib.rcParams['fonts.serif'] = ['Noto Serif', ...]
```

and then switch between serif/sans serif in a straight forward way.  You could then imagine filling these lists with a number of fonts which _may_ be installed and using the first one you find.

The current default values include a bunch of fonts that we do not ship (but expect to be on users systems with high probability).  The only fonts we ship are in https://github.com/matplotlib/matplotlib/tree/master/lib/matplotlib/mpl-data/fonts/ttf (there are a bunch of afm files as well, but those are only font metrics (glyph sizes) that we need for layout, not the full fonts).

Reply:
But this exactly creates confusions for people who're coming to our docs for setting custom fonts, like https://discourse.matplotlib.org/t/fonts-and-font-families/22132.

I'm sure an experienced developer could come up with this quick way of switching regardless of the docs. We _could_ however, add a small paragraph about where you _can_ use this type of method (giving this switching example) maybe?

It's more like saying, "You can use the non-recommended way if you know what you're doing. But if you're new to this, or if you're coming from a web-background, you should probably stick to the recommended way."

Reply:
The default values not being shipped is a bit of concern, because the statement becomes invalid (maybe I could rephrase?)